src/SymbolTable.ts

```diff
@@ -303,6 +306,23 @@ export class SymbolTable implements SymbolTypeGetter {
         return symbols;
     }
 
+
+    private cachedCircularReferenceCheck: null | boolean = null;
+
+    private hasCircularReferenceWithAncestor() {
```

Comment:
ran into some problems with circular references in symbol tables --- best way to deal with that is to check once and cache